install/install.php

```diff
@@ -328,23 +328,21 @@ public function __construct($dbh)
     );
 
     if ($success) {
-        echo "<p>" . __s('Initializing database tables and default data...') . "</p>";
-        try {
-            Toolbox::createSchema($_SESSION["glpilanguage"]);
-        } catch (\Throwable $e) {
-            echo "<p>"
-                . sprintf(
-                    __s('An error occurred during the database initialization. The error was: %s'),
-                    '<br />' . htmlescape($e->getMessage())
-                )
-                . "</p>";
-            @unlink(GLPI_CONFIG_DIR . '/config_db.php'); // try to remove the config file, to be able to restart the process
-            $prev_form($host, $user, $password);
-            return;
-        }
-        echo "<p>" . __s('OK - database was initialized') . "</p>";
+        echo "<p>" . __('Initializing database tables and default data...') . "</p>";
 
+        $alert_name = 'glpi_install_messages_container';
+        ob_start();
         $next_form();
+        $next = ob_get_clean();
+
+        echo \sprintf('<div id="%s"></div>', $alert_name);
+        echo \sprintf(
+            '<script defer>startDatabaseInstall("%s", "%s");</script>',
+            $alert_name,
+            str_replace(["\n", '"'], ['', '\\"'], $next),
+        );
+
+        $prev_form($host, $user, $password);
```

Comment:
IMHO, both forms should probably be present in the DOM when page is loaded, but hidden (and probably with disabled buttons).
Then, at the end of operation progress, the `$prev_form` should be displayed and activated if an error occurs, or the `$next_form` should be displayed and activated if the request was a success.

Reply:
Done